pkg/vsphere/tasks/waiter.go

```diff
@@ -89,11 +89,26 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
-func isTaskInProgress(err error) bool {
+const (
+	vimFault  = "vim"
+	soapFault = "soap"
+	taskFault = "task"
+)
+
+// isRetryErrors will return true for vSphere errors, which can be fixed by retry.
+// Currently the error includes TaskInProgress, NetworkDisruptedAndConfigRolledBack and InvalidArgument
+// The reason to retry for these errors is because sometimes vSphere needs longer time to finish sync up inside
+func isRetryErrors(err error) bool {
 	if soap.IsSoapFault(err) {
 		switch f := soap.ToSoapFault(err).VimFault().(type) {
 		case types.TaskInProgress:
 			return true
+		case *types.NetworkDisruptedAndConfigRolledBack:
+			logExpectedFault(soapFault, f)
+			return true
+		case *types.InvalidArgument:
```

Comment:
The error thrown in the failed case is this error, with a invalid path message.
There is one vSAN bug talking about partition, and if there is host partition, vSAN files will be only accessible to one group, and not for others. And the bug is that even vSAN suggests host to start VM, DRS or PBM might still pick up the host has no access to data to power on VM.
I don't know how the partition happens, but if that's because of network congestion, it's possible in our env.
And because the failed VM is able to be powered on some time later, it's possible to be fixed by retry.
Another reason to retry this error is because in general, all VM configuration is specified by VIC without any user input, there should not have any invalidArgument error happens unless there is system issue.